### PR TITLE
feat: change radio buttons to check box and user selection

### PR DESCRIPTION
**Pull request by DesmondTambe**
## Description

- Change the user selection option to check boxes.
- User selection functionality
- Display selected users as a horizontal scroll

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read and ran all relevant commands as specififed in the Running Tests section of the [Contributor Guide].
- [ ] The title of the PR follows the [Conventional Commits] guideline
- [ ] My local branch follows the naming standards in the [Deepsource Branch Naming Convention] or [Biodiversity Branch Naming Convention]
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy],
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

[Contributor Guide]: https://github.com/FlutterPlaza/.github/blob/main/CONTRIBUTING.md
[Conventional Commits]: https://www.conventionalcommits.org/en/v1.0.0-beta.4/
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[Biodiversity Branch Naming Convention]: https://bit.ly/3DyYSwM
[Deepsource Branch Naming Convention]: https://bit.ly/3Y08Gs4

### CODE REVIEW

File: windows/flutter/generated_plugins.cmake [hunk 1]
```diff
@@ -1,26 +0,0 @@
-#
-# Generated file, do not edit.
-#
-
-list(APPEND FLUTTER_PLUGIN_LIST
-  connectivity_plus
-  flutter_secure_storage_windows
-  local_auth_windows
```
Comment (fonkamloic) [inline windows/flutter/generated_plugins.cmake]: Why did you delete this from Git?

Comment (fonkamloic) [reply windows/flutter/generated_plugins.cmake]: Why did you delete this from Git? 

@jeffrey0606

File: windows/flutter/generated_plugin_registrant.cc [hunk 2]
```diff
@@ -1,20 +0,0 @@
...
-#include <flutter_secure_storage_windows/flutter_secure_storage_windows_plugin.h>
-#include <local_auth_windows/local_auth_plugin.h>
-
-void RegisterPlugins(flutter::PluginRegistry* registry) {
-  ConnectivityPlusWindowsPluginRegisterWithRegistrar(
-      registry->GetRegistrarForPlugin("ConnectivityPlusWindowsPlugin"));
-  FlutterSecureStorageWindowsPluginRegisterWithRegistrar(
-      registry->GetRegistrarForPlugin("FlutterSecureStorageWindowsPlugin"));
-  LocalAuthPluginRegisterWithRegistrar(
-      registry->GetRegistrarForPlugin("LocalAuthPlugin"));
```
Comment (fonkamloic) [inline windows/flutter/generated_plugin_registrant.cc]: Why did you delete this from Git? 

@jeffrey0606 

These deleted files will create missing files, and break iOS builds.

@DesmondTambe  please do not confuse, not stage changes from deleting changes. 

This delete will delete from everyone branch. 

I think you were trying to ignore the changes you made to these files. 

So I will advice using Vscode git tab to visually view all the changes you made. added the relevant one and discard the rest. 

Please never do `git add .`

**Review by fonkamloic — CHANGES_REQUESTED** [review]
Breaking changes, causing regression HERE.

File: windows/flutter/generated_plugin_registrant.cc (hunk 2 above)
Comment (DesmondTambe) [reply windows/flutter/generated_plugin_registrant.cc]: Hello @fonkamloic 

Sorry about this, I know about not using `git add .` but kinda forgot before I notice I did use it

So thought since it was generate I delete thinking it was the right thing.

Let me fix it, my bad...🤦🏾‍♂️

File: lib/home/view/widgets/user_check_select.dart [hunk 3]
```diff
@@ -66,11 +68,10 @@ class UserRadioSelect extends StatelessWidget {
             ),
           ),
 
-          // radio select button
+          // checkbox select button
           SizedBox(
-            child: Radio(
-              value: 'dezzy',
-              groupValue: username,
+            child: Checkbox(
```
Comment (jeffrey0606) [inline lib/home/view/widgets/user_check_select.dart]: Use a custom checkbox that matches the designs

File: lib/home/view/user_search_screen.dart [hunk 4]
```diff
@@ -19,27 +18,30 @@ class UserSearchScreen extends StatelessWidget {
...
+            resizeToAvoidBottomInset: false,
+            appBar: CustomAppBar(
+              titleChildWidget: CircleAvatar(
+                backgroundImage: NetworkImage('${user.photo}'),
+              ),
+              actionChildWidget: [
+                RowHeaderIcons(),
               ],
             ),
+            body: SingleChildScrollView(
```
Comment (jeffrey0606) [inline lib/home/view/user_search_screen.dart]: make the Recently sent and Contacts  section are not independently scrollable, you can achieve this with `CustomScrollView`

File: lib/home/view/widgets/search_input.dart [hunk 5]
```diff
@@ -31,22 +28,16 @@ class SearchInputWidget extends StatelessWidget {
...
           ),
-          SizedBox(
-            height: box.maxHeight * 0.02,
-          ),
+          VerticalSpacingWidget(box: box),
           Container(
             width: box.maxWidth,
-            child: FpbTextFormField(
-              key: const Key('SearchUserForPayments'),
+            child: CustomSearchInput(
```
Comment (jeffrey0606) [inline lib/home/view/widgets/search_input.dart]: Please make to use the reusable textfield we have already setup, because I think the textfield (`FpbTextFormField`) forms @TikuGaelle have are not different from what you currently have @DesmondTambe .

File: lib/home/view/widgets/user_search_list.dart [hunk 6]
```diff
@@ -1,173 +1,212 @@
 import 'package:flutter/material.dart';
```
Comment (jeffrey0606) [inline lib/home/view/widgets/user_search_list.dart:1]: This is too complex widget. Please make sure you separate them (search widget, recent contacts, and contact) into different sections and move them up to the user_search_screen

File: lib/core/presentation/widget/custom_search_input.dart [hunk 7]
```diff
@@ -0,0 +1,114 @@
+import 'package:flutter/material.dart';
```
Comment (jeffrey0606) [inline lib/core/presentation/widget/custom_search_input.dart]: Please make this `FpbTextFormField` can't satisfy your requirements before creating an new custom textfield. for more reference please have a look at my comment below on this.

**Review by jeffrey0606 — CHANGES_REQUESTED** [review]

File: lib/core/presentation/widget/custom_search_input.dart (hunk 7 above)
Comment (fonkamloic) [reply lib/core/presentation/widget/custom_search_input.dart]: You can also just wrap the FbpTextFormfield with your custom input since it has the right styling inherited from themeData

File: lib/core/presentation/widget/selected_user_item.dart [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+                    Icons.person,
+                  )
+                : Container(),
+          ),
+          Text(
+            '${username}',
+            style: TextStyle(
+              fontSize: box.maxHeight * 0.02,
+              fontWeight: FontWeight.w400,
+            ),
```
Comment (fonkamloic) [inline lib/core/presentation/widget/selected_user_item.dart]: Use style from theme

**Review by jeffrey0606 — APPROVED** [review]

**Comment by jeffrey0606** [comment]
Yo @DesmondTambe there are some conflicts here. Please resolve them and update the PR so I can merge it.

I think we should gitignore all the generated files like `app_route.gr.dart`, injection.config.dart, etc. or simply not pushing them anymore.
Please what to do you think about this @fonkamloic ?
